app/src/main/java/org/oppia/android/app/topic/revisioncard/RevisionCardViewModel.kt

```diff
@@ -32,8 +33,48 @@ class RevisionCardViewModel @Inject constructor(
     processRevisionCardLiveData()
   }
 
-  fun clickReturnToTopic(@Suppress("UNUSED_PARAMETER") v: View) {
-    returnToTopicClickListener.onReturnToTopicClicked()
+  private val topicLiveData: LiveData<AsyncResult<EphemeralTopic>> by lazy {
+    getTopicResultLiveData()
+  }
+
+  private fun getTopicResultLiveData(): LiveData<AsyncResult<EphemeralTopic>> {
+    return topicController.getTopic(profileId, topicId).toLiveData()
+  }
+
+  val nextSubtopicLiveData: LiveData<EphemeralSubtopic> by lazy {
+    Transformations.map(topicLiveData, ::processNextSubtopicData)
+  }
+
+  val previousSubtopicLiveData: LiveData<EphemeralSubtopic> by lazy {
+    Transformations.map(topicLiveData, ::processPreviousSubtopicData)
+  }
+
+  private fun processPreviousSubtopicData(
+    topicLiveData: AsyncResult<EphemeralTopic>
+  ): EphemeralSubtopic? {
```

Comment:
For this & the next: do they ever return null? It doesn't seem like it.

Reply:
done